plugin/trino-hive/src/main/java/io/trino/plugin/hive/HivePartitionManager.java

```diff
@@ -199,8 +214,21 @@ public HiveTableHandle applyPartitionResult(HiveTableHandle handle, HivePartitio
 
     public List<HivePartition> getOrLoadPartitions(SemiTransactionalHiveMetastore metastore, HiveTableHandle table)
     {
+        // In case of partitions not being loaded, their permissible values are specified in `HiveTableHandle#getCompactEffectivePredicate,
+        // so we do an intersection of getCompactEffectivePredicate and HiveTable's enforced constraint
+        TupleDomain<ColumnHandle> summary = table.getEnforcedConstraint().intersect(
+                table.getCompactEffectivePredicate()
+                        .transformKeys(ColumnHandle.class::cast));
```

Comment:
Having something like https://github.com/trinodb/trino/pull/11078 would allow to `intersect` without copying the other `TupleDomain` in `transformKeys`.

Reply:
Would be happy to rebase it on top that PR. Once it is merged.